android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BluetoothSite.java

```diff
@@ -193,9 +198,22 @@ public void onServicesDiscovered(BluetoothGatt gatt, int status) {
     }
 
     characteristic = gatt.getService(SERVICE_UUID).getCharacteristic(CHARACTERISTIC_WEBPAGE_UUID);
+    checkWebpageGzipped();
+    
     gatt.readCharacteristic(characteristic);
   }
 
+  private void checkWebpageGzipped() {
```

Comment:
OK, then same descriptor I guess.  You could try unzipping it and see if it fails.  If that's the case, it's not gzipped.